src/main/java/com/snowflake/kafka/connector/internal/IngestSdkProvider.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023 Snowflake Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.snowflake.kafka.connector.internal;
+
+import com.snowflake.kafka.connector.internal.streaming.StreamingUtils;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+public class IngestSdkProvider {
+  private static final String STREAMING_CLIENT_PREFIX_NAME = "KC_CLIENT_";
+
+  private LoggerHandler LOGGER;
+  private int streamingIngestClientCount;
+  private SnowflakeStreamingIngestClient streamingIngestClient;
+
+  /**
+   * This is a wrapper for the ingest sdk to help define a contract between KC and Ingest
+   *
+   * <p>Ideally all ingest sdk calls will go through this to make mocking and integration tests
+   * easier to write (potentially also to run most of the integration tests without needing a
+   * snowflake connection). Currently it only manages the client.
+   */
+  public IngestSdkProvider() {
+    LOGGER = new LoggerHandler(this.getClass().getName());
+    this.streamingIngestClientCount = 0;
+  }
+
+  public SnowflakeStreamingIngestClient createStreamingClient(
```

Comment:
If you are leaning towards a static instance of client, we can simply replace this with private method and expose a public method which returns the singleton.

Reply:
This will make testing a lot simpler.

Reply:
I am not leaning towards a static client, however we could make IngestSdkProvider a static singleton by doing what you said - make createStreamingClient() private and only use getStreamingClient. 

Currently I'm leaning towards keeping it as is, because of the uncertainty around if connector.start or task.start will be called first. For context, the earliest the client can be created is connector.start, but task.start also needs the client. The static ingestSdkProvider is instantiated in connector.initialize() which is guaranteed to be the first method called. That way when task.start() grabs the ingestSdkProvider, it will not be null. 

Essentially if we can confirm that task.start always and will always happen after connector.start is completed, then I'd be happy to make ingestSdkProvider a singleton

Reply:
Oops, lemme walk that back - forgot Java is pass by value not pass by reference. I'll think more on this

Reply:
This thread is pretty similar to the following one - lets converge by talking on that one only

https://github.com/snowflakedb/snowflake-kafka-connector/pull/528#discussion_r1088807756

Reply:
Ok I went with the second way described above. My one concern is that setting the global static IngestSdkProvider may persist cross test and cause test flakiness. Remembering to set the provider before tests is a solution, but not the cleanest